### PR TITLE
Add "recent" swaps section to token to buy list

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1635

## What changed (plus any additional context for devs)
Added a new section that shows the last three user swaps per chain. These recents prioritize favorites and bridge assets if they exist in those lists.

## Screen recordings / screenshots
Screenshot 2024-07-25 at 12 04 27 PM
Screenshot 2024-07-25 at 12 04 35 PM

### CODE REVIEW

File: src/state/swaps/swapsStore.ts [hunk 1]
```diff
@@ -80,15 +149,34 @@ export const swapsStore = createRainbowStore<SwapsState>(
...
+      latestSwapAt.set(chainId, now);
+
+      set({ recentSwaps: new Map(recentSwaps) });
+    },
   }),
   {
     storageKey: 'swapsStore',
-    version: 1,
-    // NOTE: Only persist the settings
+    version: 3,
```
Comment (walmat) [inline src/state/swaps/swapsStore.ts:170]:
```suggestion
    version: 2,
```

File: src/__swaps__/types/swap.ts [hunk 2]
```diff
@@ -21,3 +24,7 @@ export interface RequestNewQuoteParams {
   lastTypedInput: inputKeys;
   outputAmount: inputValuesType['outputAmount'];
 }
+
+export type RecentSwap = {
+  swappedAt: number;
+} & ExtendedAnimatedAssetWithColors;
```
Comment (walmat) [inline src/__swaps__/types/swap.ts:30]: Do we want more metadata here?

File: src/state/swaps/swapsStore.ts [hunk 3]
```diff
@@ -34,6 +36,74 @@ export interface SwapsState {
   setSource: (source: Source | 'auto') => void;
   degenMode: boolean;
   setDegenMode: (degenMode: boolean) => void;
+
+  // recent swaps
+  latestSwapAt: Map<ChainId, number>;
```
Comment (greg-schrammel) [inline src/state/swaps/swapsStore.ts:41]: is this prop used? do we need it?

**Review by greg-schrammel — APPROVED** [review]

File: src/state/swaps/swapsStore.ts (hunk 3 above)
Comment (walmat) [reply src/state/swaps/swapsStore.ts:41]: Not used but was in spec: https://www.notion.so/rainbowdotme/Show-Popular-in-Rainbow-Tokens-in-Swaps-on-App-BX-cc7b4a27f398427f8ede707c22943d94?pvs=4#eec96f7511ab4d3a9060a0980e14e787

**Pull request merged**